### PR TITLE
feat: add power meter instrument

### LINKED ISSUE #91: Implement a basic power-meter instrument
## Context
The simulator currently has visualization and specialized analysis tools, but the planned automated test-flow / ATP work needs a small, machine-readable measurement primitive. A power meter is the most useful first instrument because it provides a simple scalar observable for gain, attenuation, compression, filtering, and chain-level tests.

## Scope
Implement a basic power-meter measurement facility, preferably with measurement logic separated from the ImGui widget so it can be used by both the GUI and automated test flows.

The initial version should support:
- Selecting an output pin or signal node as the measurement source.
- Measuring total signal power in dBm.
- Clearly defined handling of tones, noise, or both.
- A validity/error state for unavailable or invalid input.
- A small GUI instrument panel or equivalent user-facing display.
- Unit tests for the measurement calculations and edge cases.

The design should leave room for later additions such as measurement bandwidth, channel power, tone selection, peak/average detector modes, and power-versus-frequency sweeps, without requiring those features in the first increment.

## Measurement contract
Document:
- The power reference and dBm conversion.
- Whether the initial result includes tones, integrated noise, or both.
- How multiple tones are combined.
- How an empty spectrum, invalid values, and missing sources are reported.
- Whether the result is instantaneous, frame-based, or averaged.

## Acceptance criteria
- A selected simulator output can be measured as a scalar power value in dBm.
- The result is available through a core/testable API rather than only through rendered UI text.
- Tone-only and noise-inclusive behavior is covered by focused tests.
- Empty, missing, and invalid input cases do not produce misleading numeric results.
- The GUI exposes the basic measurement and its source selection, if a GUI panel is included in the implementation.
- The measurement can be consumed by a future ATP/test-flow implementation without depending on ImGui.

## Related work
- ATP/test-flow harness: #87
- Measured-data replay and comparison: #89
- Existing specialized instrument: Network Analyzer

### PR DESCRIPTION
**Pull request by striderZA**
## Summary

Adds a machine-readable power-meter instrument with a GUI output-pin picker. The measurement engine sums tone and integrated noise power in watts, converts the result to dBm, and reports explicit invalid-input states.

## Related issue

Closes #91

## Type of change

- [ ] Bug fix
- [x] New feature
- [ ] Refactor
- [ ] Documentation
- [x] Build / CI

## Test plan

- [x] `cmake --build build` — passed
- [ ] `ctest --test-dir build --output-on-failure` — not run; focused coverage was run per repository guidance
- [x] `ctest --test-dir build -R '^test_power_meter$' --output-on-failure` — 1/1 passed
- [x] Clean CMake configure — passed
- [x] Direct clang-format 18 check on all changed C++ files — passed
- [ ] Manual verification steps (if applicable)

## Checklist

- [x] Code follows the existing style
- [x] Changed C++ files were checked with clang-format 18
- [x] Added standalone regression tests for calculations and invalid inputs
- [x] Complete project build passed

### CODE REVIEW

File: app/src/app.cpp [hunk 1]
```diff
@@ -442,6 +445,7 @@ void RfSimulatorApp::saveProject(const std::string &path) {
 }
 
 void RfSimulatorApp::loadProject(const std::string &path) {
+    m_power_meter_widget->clearSource();
```
Comment (striderZA) [inline app/src/app.cpp]: Required: `clearSource()` runs before `ProjectSerializer::load()`. For a missing/unreadable or invalid-JSON file, `load()` returns false without resetting the current project (project_serializer.cpp:329-355), so a failed Open leaves the graph intact but silently loses the meter selection. Clear this session-only selection only when the project state was actually reset/replaced (while retaining the clear on `newProject()`); expose/propagate the load outcome if needed so reset-on-invalid-shape/restore-failure paths still cannot leave a stale pin ID. Please add a regression test for a failed load preserving the existing project and meter selection.

Comment (striderZA) [reply app/src/app.cpp]: Fixed in 88a8d6e. ProjectSerializer now reports whether a load reset project state; RfSimulatorApp preserves the Power Meter source on non-mutating failures, clears it after successful loads, and clears it when a failed load already reset state. Added test_power_meter_app coverage for both missing-file preservation and reset-on-invalid-shape clearing. Verified with clang-format on changed C++ files and ctest: test_power_meter, test_power_meter_app, test_issue48_json_loader, and test_issue77_save_failure all passed.
